File: RuntimeSystem-Benchmarking/Code/Linear Reg R/simple_linear_regression.R
# Simple Linear Regression

# Importing the dataset
n <- 100
T <- matrix(0, n, 4)
iter <- 1
data <- list('/Users/katieturnlund/Workspace/CS263_Jaber_Katie/Code/Linear Reg R/Data_Salary100.csv',
 '/Users/katieturnlund/Workspace/CS263_Jaber_Katie/Code/Linear Reg R/Data_Salary1k.csv',
 '/Users/katieturnlund/Workspace/CS263_Jaber_Katie/Code/Linear Reg R/Data_Salary10k.csv',
 '/Users/katieturnlund/Workspace/CS263_Jaber_Katie/Code/Linear Reg R/Data_Salary100k.csv')
 for (item in data) {
  for (i in 1:n) {
    start <- Sys.time()
    dataset <- read.csv(item)

    # Splitting the dataset into the Training set and Test set
    # install.packages('caTools')
    library(caTools)
    set.seed(123)
    split = sample.split(dataset$Salary, SplitRatio = 2/3)
    training_set = subset(dataset, split == TRUE)
    test_set = subset(dataset, split == FALSE)

    # Feature Scaling
    # training_set = scale(training_set)
    # test_set = scale(test_set)

    # Fitting Simple Linear Regression to the Training set
    regressor = lm(formula = Salary ~ YearsExperience,
                  data = training_set)

    # Predicting the Test set results
    y_pred = predict(regressor, newdata = test_set)

    # Visualising the Training set results
    library(ggplot2)
    ggplot() +
      geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
                colour = 'red') +
      geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
                colour = 'blue') +
      ggtitle('Salary vs Experience (Training set)') +
      xlab('Years of experience') +
      ylab('Salary')

    # Visualising the Test set results
    library(ggplot2)
    ggplot() +
      geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
                colour = 'red') +
      geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
                colour = 'blue') +
      ggtitle('Salary vs Experience (Test set)') +
      xlab('Years of experience') +
      ylab('Salary')
      end <- Sys.time()
    T[i, iter] <- end - start
  }
  iter <- iter + 1
 }

library(MASS)
write.matrix(T,file="/Users/katieturnlund/Workspace/CS263_Jaber_Katie/Code/Linear Reg R/R_results_linReg.csv")

File: RuntimeSystem-Benchmarking/Code/Linear Reg Python/simple_linear_regression.py
# Simple Linear Regression


# Importing the libraries

import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import time
from tqdm import tqdm

n = 100
T = np.zeros((n, 4))
iter = 0
data = ['Data_Salary100.csv', 'Data_Salary1k.csv', 'Data_Salary10k.csv', 'Data_Salary100k.csv']
for item in tqdm(data, position = 0):
    for i in tqdm(range(n), position=1):
        start_time = time.time()
        # Importing the dataset
        dataset = pd.read_csv(item)
        X = dataset.iloc[:, :-1].values
        y = dataset.iloc[:, -1].values

        # Splitting the dataset into the Training set and Test set
        from sklearn.model_selection import train_test_split
        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

        # Training the Simple Linear Regression model on the Training set
        from sklearn.linear_model import LinearRegression
        regressor = LinearRegression()
        regressor.fit(X_train, y_train)

        # Predicting the Test set results
        y_pred = regressor.predict(X_test)
        
        # Visualising the Training set results
        plt.scatter(X_train, y_train, color = 'red')
        plt.plot(X_train, regressor.predict(X_train), color = 'blue')
        plt.title('Salary vs Experience (Training set)')
        plt.xlabel('Years of Experience')
        plt.ylabel('Salary')
        #plt.savefig('test1.png')

        # Visualising the Test set results
        plt.scatter(X_test, y_test, color = 'red')
        plt.plot(X_train, regressor.predict(X_train), color = 'blue')
        plt.title('Salary vs Experience (Test set)')
        plt.xlabel('Years of Experience')
        plt.ylabel('Salary')
        #plt.savefig('test2.png')

        T[i, iter] = time.time() - start_time
    iter += 1

df = pd.DataFrame(T, columns = ['100', '1k', '10k', '100k'])
df.to_csv('python_results_linReg.csv', header = False, index = False)